File: schema/menu.py
"""菜单树响应 schema。

`MenuNodeOut` 是递归模型：每个节点可挂 `children`。ID 字段统一走
`IdStr` / `IdStrNonNull`，序列化到 JSON 时变成字符串以避免 JS
`Number.MAX_SAFE_INTEGER` 精度截断。
"""
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field

from schema._types import IdStr, IdStrNonNull


class MenuNodeOut(BaseModel):
    """单个菜单节点 + 子节点。

    - `path` 为 NULL 表示分组节点（前端渲染为 <el-sub-menu>）。
    - `parent_id` 顶层节点为 NULL。
    - `icon` 是 Element-Plus 图标组件名（如 'House'）；前端按名查表渲染。
    """

    id: IdStrNonNull
    parent_id: IdStr = None
    code: str
    title: str
    path: str | None = None
    icon: str | None = None
    sort_order: int
    children: list["MenuNodeOut"] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)


MenuNodeOut.model_rebuild()

File: frontend/src/types/menu.ts
/** 后端 MenuNodeOut 对齐 */
export interface MenuNode {
  id: string
  /** 乐观锁版本号；每次 UPDATE 自增 */
  version: number
  parent_id: string | null
  code: string                    // 稳定业务键（如 'parts_list'）；路由 meta.menuCode 与之对齐
  title: string
  path: string | null             // null = 分组节点（前端渲染 <el-sub-menu>）
  icon: string | null              // Element-Plus 图标组件名（如 'House'）
  sort_order: number
  children: MenuNode[]
}